src/libsystemd/sd-json/json-util.c

```diff
@@ -124,6 +124,22 @@ int json_dispatch_in_addr(const char *name, sd_json_variant *variant, sd_json_di
         return 0;
 }
 
+int json_dispatch_in_addr_union(const char *name, sd_json_variant *variant, sd_json_dispatch_flags_t flags, void *userdata) {
```

Comment:
hmm, i am not sure how i feal about this one. 

i.e. the size of this should really verified whether it matches the expected address size of the family. But that's a check that only can be done once the full set of parameters is processed.

This could be used for bad stuff otherwise, for example in this current version if you pass only 4 bytes of address but pick ipv6 as address format this would happily accept that and then later run into uninitialized memory...

Reply:
I took the suggestion to use the iovec dispatcher instead.